### DISCUSSION TITLE
How to implement transport binding behind a proxy with ssl termination

### DISCUSSION THREAD

**Discussion by oruchreis** [CoreWCF/CoreWCF#1024] [Q&A]
Hi,
I'm trying to run a CoreWCF project behind Azure Application Gateway(appGw) with ssl termination(offloading) with kubernetes backend. The service exposes 80 port to the appGw proxy, and appGw exposes 443 port to public. I configured the x-forwarded-* headers in appGw, and configured the `UseForwardedHeaders` middleware in the startup of core wcf project. I've tested that the application can get x-forwarded-* headers succesfully. But when I try to send a soap request with Basic or WS http binding, the appGw sends the request to http port, and the application logs the request as http://** url. And the service can't find the endpoint and throws 404 not found error.

We are using `BasicHttpBinding` and `WsHttpBinding` `Mtom` with transport security with `ClientCredentionType` as `None`. And our clients using the service with these settings, and we can't request our clients to change their binding settings right now. If I set the `SecurityMode` to `None` instead of transport, will our clients must change the binding settings? When I add another `ServiceEndpoint` with `SecurityMode.None`, then the wsdl shows two service address **http** and **https**. But I need to expose and show only **https** urls.

Our binding settngs here:
```csharp
.AddServiceEndpoint(serviceMetaData.Type, contract,
new BasicHttpBinding(BasicHttpSecurityMode.Transport)
{
    MaxReceivedMessageSize = int.MaxValue,
    MaxBufferSize = int.MaxValue,
    Security = new BasicHttpSecurity
    {
        Mode = BasicHttpSecurityMode.Transport,
        Message = new BasicHttpMessageSecurity { ClientCredentialType = BasicHttpMessageCredentialType.Certificate },
        Transport = new HttpTransportSecurity { ClientCredentialType = HttpClientCredentialType.None }
    }
},
new Uri($"/soap", UriKind.Relative), null,
serviceEndpoint =>
{
    serviceEndpoint.EndpointBehaviors.Add(new DefaultBehavior());
});
.AddServiceEndpoint(serviceMetaData.Type, contract,
new WSHttpBinding(SecurityMode.TransportWithMessageCredential)
{
    MaxBufferPoolSize = long.MaxValue,
    MaxReceivedMessageSize = int.MaxValue,
    MessageEncoding = WSMessageEncoding.Mtom,
    Security = new WSHTTPSecurity
    {
        Mode = SecurityMode.Transport,
        Message = new NonDualMessageSecurityOverHttp { ClientCredentialType = MessageCredentialType.Certificate },
        Transport = new HttpTransportSecurity { ClientCredentialType = HttpClientCredentialType.None }
    }
},
new Uri($"/", UriKind.Relative), null,
serviceEndpoint =>
{
    serviceEndpoint.EndpointBehaviors.Add(new DefaultBehavior());
});
```

**Comment by mclea** [comment]
Related?:

https://github.com/CoreWCF/CoreWCF/issues/659#issuecomment-1125226066

**Reply by a-tze** [reply]
Unfortunately, this does not cover one specific case: dual-protocol endpoints behind a reverse proxy. A good solution would be to (optionally) reflect all X-forwarded-* headers into the location attribute in the generated WSDL.
